Android.mk

```diff
@@ -0,0 +1,66 @@
+#
+# vkrunner
+#
+# Copyright (C) 2018 Google LLC
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice (including the next
+# paragraph) shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+#
+
+LOCAL_PATH := $(call my-dir)
+
+$(call generate-file-dir,$(LOCAL_PATH)/config.h)
+$(LOCAL_PATH)/config.h: $(LOCAL_PATH)/config.h.android
```

Comment:
What depends on $(LOCAL_PATH)/config.h ?
Changing config.h.android should trigger rebuilding all the .c source files that depend on config.h but I don't think that dependency is captured anywhere.

Reply:
Right .. this is the part I hate Android NDK.
If you see [Android.mk document](http://android.mk/), it says "The build system handles many details for you. For example, you don't need to list header files or explicit dependencies between
  generated files in your Android.mk. The NDK build system will compute these automatically for you.".

If we use its `LOCAL_SRC_FILES`, there is no way to explicitly specify dependencies.
It also says "The LOCAL_SRC_FILES variables must contain a list of C and/or C++ source files that will be built and assembled into a module. Note that you should
**not list header and included files** here, because the build system will compute dependencies automatically for you; just list the source files
that will be passed directly to a compiler, and you should be good.".

To specify the explicit dependencies, we must use cmake instead of Android.mk.

Reply:
Ok, if it computes the header dependency on its own then I'm fine.

CMake does the same thing, by using flags -MD and -MT flags on the first compile to capture the  dependencies actually used to compile.

Reply:
Sorry I was wrong. I think I can put
```
config.h: config.h.android
    cp config.h.android config.h
XXX.c: config.h
```

I updated Android.mk. Please take a look.